src/Symfony/Component/Security/Core/Security.php

```diff
@@ -46,6 +46,11 @@ public function getUser()
             return null;
         }
 
+        if (!$user instanceof UserInterface) {
+            @trigger_error(sprintf('Accessing the user object "%s" that is not an instance of "%s" from "%s()" is deprecated since Symfony 4.2, use "getToken()->getUser()" instead.', get_class($user), UserInterface::class, __METHOD__), E_USER_DEPRECATED);
```

Comment:
`$user` may not be an object here: `is_object($user) ? get_class($user) : $user`